### PR TITLE
Pr feedback on scilla integration

### PR DESCRIPTION
**Pull request by n-hutton**

### CODE REVIEW

File: scilla/src/backend_collector.rs [hunk 1]
```diff
@@ -61,27 +61,26 @@ impl<'a, B: Backend> BackendCollector<'a, B> {
...
+            let account = Account {
+                nonce: self.backend.basic(address).nonce.as_u64(),
+                code: self.backend.code(address),
+                storage_root: None,
+                is_scilla: true,
+            };
+
+            e.insert(Some((account, HashMap::from([(key, value)]))));
+        } else {
+            let entry = self.account_storage_buffered.get_mut(&address).unwrap();
```
Comment (JamesHinshelwood) [inline scilla/src/backend_collector.rs:76]: Why did you change from `match` to `if let`? In my opinion, it was clearer before because you handle both cases explicitly rather than defaulting into the `else` and then having to `.unwrap()` once you know the key is there.

**Review by JamesHinshelwood — DISMISSED** [review]

File: scilla/src/backend_collector.rs (hunk 1 above)
Comment (n-hutton) [reply scilla/src/backend_collector.rs:76]: Perhaps I misunderstood your original comment - I thought this was what you specified. I will switch it to the match again

**Review by JamesHinshelwood — DISMISSED** [review]
